src/CommonITILObject.php

```diff
@@ -976,32 +976,66 @@ public function canApprove()
+            )
+            || (
+                Session::haveRight("followup", ITILFollowup::ADD_AS_OBSERVER)
+                && $this->isUser(CommonITILActor::OBSERVER, Session::getLoginUserID())
+            )
+            || Session::haveRight('followup', ITILFollowup::ADDALLITEM)
+            || (
+                Session::haveRight('followup', ITILFollowup::ADD_AS_GROUP)
+                && isset($_SESSION["glpigroups"])
+                && $this->haveAGroup(CommonITILActor::REQUESTER, $_SESSION['glpigroups'])
+            )
+            || $this->isUser(CommonITILActor::ASSIGN, Session::getLoginUserID())
+            || (
+                isset($_SESSION["glpigroups"])
+                && $this->haveAGroup(CommonITILActor::ASSIGN, $_SESSION['glpigroups'])
             )
-         )
-         || (
-            Session::haveRight("followup", ITILFollowup::ADD_AS_OBSERVER)
-            && $this->isUser(CommonITILActor::OBSERVER, Session::getLoginUserID())
-         )
-         || Session::haveRight('followup', ITILFollowup::ADDALLTICKET)
-         || (
-            Session::haveRight('followup', ITILFollowup::ADDGROUPTICKET)
-            && isset($_SESSION["glpigroups"])
-            && $this->haveAGroup(CommonITILActor::REQUESTER, $_SESSION['glpigroups'])
-         )
-         || $this->isUser(CommonITILActor::ASSIGN, Session::getLoginUserID())
-         || (
-            isset($_SESSION["glpigroups"])
-            && $this->haveAGroup(CommonITILActor::ASSIGN, $_SESSION['glpigroups'])
-         )
-         || $this->isUserValidationRequested(Session::getLoginUserID(), true)
+            || $this->isUserValidationRequested(Session::getLoginUserID(), true)
+        );
+    }
+
+    public function canAddTasks()
+    {
+        return (
+             (
+                 Session::haveRight(CommonITILTask::$rightname, CommonITILTask::ADDMY)
+                 && (
+                     $this->isUser(CommonITILActor::REQUESTER, Session::getLoginUserID())
+                     || (
+                         isset($this->fields["users_id_recipient"])
+                         && ($this->fields["users_id_recipient"] == Session::getLoginUserID())
+                     )
+                 )
+             )
+             || (
+                 Session::haveRight(CommonITILTask::$rightname, CommonITILTask::ADD_AS_OBSERVER)
+                 && $this->isUser(CommonITILActor::OBSERVER, Session::getLoginUserID())
+             )
+             || Session::haveRight(CommonITILTask::$rightname, CommonITILTask::ADDALLITEM)
+             || (
+                 Session::haveRight(CommonITILTask::$rightname, CommonITILTask::ADD_AS_GROUP)
+                 && isset($_SESSION["glpigroups"])
+                 && $this->haveAGroup(CommonITILActor::REQUESTER, $_SESSION['glpigroups'])
+             )
+             || (
+                 Session::haveRightsOr(CommonITILTask::$rightname, [CommonITILTask::ADD_AS_TECHNICIAN, UPDATE, static::class::READMY])
```

Comment:
We discussed about if with @orthagh and we thought this previously existing condition, makes no sense. We consider that, as long as you are assigned, you should be able to add tasks, even if you have not the `READMY` right.

```suggestion
                 Session::haveRightsOr(CommonITILTask::$rightname, [CommonITILTask::ADD_AS_TECHNICIAN, UPDATE])
```